app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -168,9 +168,10 @@ class WireActivityViewModel @Inject constructor(
 
     fun navigationArguments() = navigationArguments.values.toList()
 
-    fun startNavigationRoute(navigationItem: NavigationItem? = null): String = when {
+    fun startNavigationRoute(navigationItem: NavigationItem? = null, isSharingIntent: Boolean = false): String = when {
         shouldGoToMigration() -> NavigationItem.Migration.getRouteWithArgs()
```

Comment:
can we do something similiar to 
shouldGoToWelcome() 
shouldGoToMigration()  ? Instead of passing a flag, flags are hard to understand I think :S

Reply:
We need to pass the flag to the VM, otherwise we would need to pass the reference to the activity to check whether there is a sharing intent, coupling it unncessarily with it and increasing the chances of a memory leak. 
But if you have any specific code suggestion please send it my way :)

Reply:
lets go with this appoach for now no suggestions in my mind at the moment :S